### ISSUE TITLE
Translations with different system of numbering ayats (ayats-combined).

### ISSUE THREAD

**Issue by Al-Muhandis** [quranacademy/QuranDataExchange#1]
[Wiki page](https://github.com/alquran-foundation/DataExchangeProject/wiki)
CSV/TSV-files for what? To present quran translation content (ayat by ayat) or to present quran meta data?

**Comment by rguliev** [comment]
Assalamu aleykum!
For all table-like data that will be stored. For now there is no data but, in shaa Allah, it will word by word Quran, word by word translations, maybe something like in corpus Quran, and other.

Can we close this?

**Comment by Al-Muhandis** [comment]
Wa aleikumu ssalam wa rahmatuLlah. I know that there are no data available. Just keep in mind also that there are some translations in which another system of numbering ayats and most importantly, that some ayats-combined, for example, in a similar forms: _2:17-19. ..._ .
And Yes, You can close this

**Comment by rguliev** [comment]
Thank you for informing. This is very important note. We will take care of it, in shaa Allah. I leave this issue open until we get solution for such translations.

**Comment by mustafa0x** [comment]
This can be solved in a number of ways; my approach is to have the following ayahs reference the first ayah.

E.g.

```
"1": "<translation of verses 1-3>",
"2": "1",
"3": "1",
```

**Comment by Al-Muhandis** [comment]
I think that from the principle of minimalism, it can simply **not specify** the following verses. If we take the basis such CSV as [this](https://raw.githubusercontent.com/alquran-foundation/translations/master/Nepali/ne_islami_sangh_nepal/ne_islami_sangh_nepal.csv) then we can just **skip** the corresponding verses, as the combination can only neighbouring verses and in the above CSV format specify the numbers of the verses. That is, for example, russian [al-muntahab translation](https://falaq.ru/quran/munt/107:2-5) then it should be presented in the following CSV format:
```
107,2, "Если хочешь знать, кто он, то ведь он тот, кто гонит сироту жестоко, унижает и притесняет его и не призывает к тому, чтобы накормить бедняков."
107,4, "Гибель тем молящимся, которые обладают вышеуказанными качествами и небрежно совершают свои молитвы, которые не приносят им никакой пользы, "
```

**Comment by mustafa0x** [comment]
That'll lead to ambiguity when a translation doesn't translate certain ayahs. A rare case maybe, but it definitely happens.

**Comment by Al-Muhandis** [comment]
I understand that there is a small amount of ambiguity, but I want to note that this is still like the source code, with a specified and prescribed structure. After all, it does not happen that ayah translations can be skipped or happens? Only when the case of a combination of verses as in this case. In addition, in this case, entering the number of ayah in the translation field, there is also a small amount of ambiguity, as if the reference-the number of ayah is the translation of this ayah. IMHO

**Comment by mustafa0x** [comment]
> but I want to note that this is still like the source code, with a specified and prescribed structure

I'm sorry, I don't follow. =)

>  all, it does not happen that ayah translations can be skipped or happens?

Yeah, that's the issue. So, in your example:

```
107,2, "Если хочешь знать, кто он, то ведь он тот, кто гонит сироту жестоко, унижает и притесняет его и не призывает к тому, чтобы накормить бедняков."
107,4, "Гибель тем молящимся, которые обладают вышеуказанными качествами и небрежно совершают свои молитвы, которые не приносят им никакой пользы, "
```

We don't know — is ayah 3 part of ayah 2, or was it not translated?

**Comment by mustafa0x** [comment]
Also, that a key exists is needed. For instance, if the user clicks on translation for ayah 3, our query for ayah 3 will return no results.

For example: `SELECT * FROM ayahs WHERE verse = '107,3';`.

**Comment by mustafa0x** [comment]
> entering the number of ayah in the translation field, there is also a small amount of ambiguity, as if the reference-the number of ayah is the translation of this ayah

Yes, in this case we are using the content field for two purposes: as content or as a reference. However, this overloading doesn't really present any issues — a simple check easily distinguishes.

**Comment by Al-Muhandis** [comment]
> We don't know — is ayah 3 part of ayah 2, or was it not translated?

Then I'm sorry, I misunderstood You. That is, in reality there are such translations, where the verses are simply missed?! In this case, of course the method I have specified is not suitable. Then I agree with You

**Comment by mustafa0x** [comment]
> That is, in reality there are such translations, where the verses are simply missed?

There are many incomplete or partial translations. For some rare languages, the QPC only translates al-Fatiha and Juz' Amm. Thus, our spec should be able to handle such situations.

Also, if we start including commentaries, then we will no doubt have issues. For instance, al-jalalayn doesn't have commentary on every ayah, so a translation of al-jalalayn will have the same issue.

**Comment by mustafa0x** [comment]
بارك الله فيكم

**Comment by Al-Muhandis** [comment]
@mustafa0x , 
و إياك!
